tests/testthat/test-model-compile.R

```diff
@@ -190,3 +190,23 @@ test_that("compile errors are shown", {
     "An error occured during compilation! See the message above for more information."
   )
 })
+
+test_that("dir arg works for cmdstan_model", {
+  skip_on_cran()
+  tmp_dir <- tempdir()
+  mod_dir <- cmdstan_model(stan_program, dir = tmp_dir)
+  expect_equal(dirname(mod_dir$exe_file()), tmp_dir)
+  expect_true(file.exists(mod_dir$exe_file()))
```

Comment:
It doesn't really matter but since we're already using checkmate we could use (here and below)

```
checkmate::expect_file_exists(...)
```

instead of 

```
testthat::expect_true(file.exists(...))
```